### PR TITLE
Update analyzer rulesets

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: eng/CodeAnalysis.ruleset [hunk 1]
```diff
@@ -174,8 +174,9 @@
     <Rule Id="CA2247" Action="Warning" />       <!-- Argument passed to TaskCompletionSource constructor should be TaskCreationOptions enum instead of TaskContinuationOptions enum. -->
     <Rule Id="CA2248" Action="Warning" />       <!-- Provide correct 'enum' argument to 'Enum.HasFlag' -->
     <Rule Id="CA2249" Action="Warning" />       <!-- Consider using 'string.Contains' instead of 'string.IndexOf' -->
-    <Rule Id="CA2250" Action="Warning" />       <!-- Use 'ThrowIfCancellationRequested' -->
+    <Rule Id="CA2250" Action="None" />          <!-- Use 'ThrowIfCancellationRequested' -->
     <Rule Id="CA2251" Action="Warning" />       <!-- Use 'string.Equals' -->
+    <Rule Id="CA2252" Action="Warning" />       <!-- This API requires opting into preview features -->
```
Comment (pgovind) [inline eng/CodeAnalysis.ruleset:179]: Change this to None? Once I get feedback from Tanner and iterate on this a little more, we can turn it back to Warning?

Comment (stephentoub) [reply eng/CodeAnalysis.ruleset:179]: This was supposed to be None.

**Comment by xtqqczze** [comment]
@stephentoub  CA2252 This API requires opting into preview features

**Comment by stephentoub** [comment]
> Should CA2252 be in Microsoft.CodeAnalysis.NetAnalyzers

Yes, it's correct.

> I'm not seeing it in 6.0.0-rc1.21366.3

This repo is currently using:
https://github.com/dotnet/runtime/blob/f100379cb705322bf0a47680cc9220f069ed3754/eng/Versions.props#L53
and it's there:
image

**Comment by pgovind** [comment]
So I was speaking to the analyzer repo folks and was informed that the DetectPreviewFeatureAnalyzer failed to flow to the sdk's P7 branch because of a transient Nuget failure. https://github.com/dotnet/sdk/pull/19084 just popped up. With that PR, I assume the analyzer will flow to the P7 branch, and we then risk subsequent P7 branch runtime builds failing? I wonder if we should be safe here and backport this PR just in case?

/backport to release/6.0-preview7

**Pull request merged**